pkg/agent/agent.go

```diff
@@ -216,14 +216,6 @@ func (k *Klocksmith) watchUpdateStatus(update func(s updateengine.Status)) {
 
 	go k.ue.ReceiveStatuses(ch, stop)
 
-	// synthesize first message
```

Comment:
To be more clear, the issue you're solving is that `ch` has been handed off to a goroutine (which may possibly send to that channel). Then we get the status directly and proceed to write it to the channel ourselves which blocks if something was already sent to the channel.

I'm not sure how the original author envisioned this code. Perhaps they intended it without `ch <- st`, the GetStatus call. Still seems odd to me even after your change, since one call is made to GetStatus.